test/controllers/password_credentials_controller_test.rb

```diff
@@ -0,0 +1,66 @@
+require "test_helper"
+
+class PasswordCredentialsControllerTest < ActionDispatch::IntegrationTest
+  include ActionDispatch::TestProcess::FixtureFile
+
+  setup do
+    people(:keith_registered)
+    @user = users(:keith)
+    credentials(:keith_password)
+
+    @settings = {
+      email_from: "teampeople@example.com",
+      product_name: "Product Name"
+    }
+    @features = {
+      password_reset_email: true,
+      email_sender_postmark: true
+    }
```

Comment:
You probably want to do the stubbing of settings and features in here, right? Because it's not currently being applied to all of the tests but we probably want it to be activated for all of the tests to ensure they're fully running as we'd expect them to.

In terms of syntax, I lean towards not having `stub_settings_and_features` read the instance var. That's a little magical to set an instance var and then know that a method is reading it. I think it's better to explicitly pass it in. My inclination would be do:

Oh, but shoot, I think the reason you didn't stub this within `setup do` is because the method requires a block.

How about this for a solution — both for this controller test and the other controller test:

1) We drop stub_settings_and_features and instead add a couple new methods to that helper (chatgpt wrote this for me, I didn't test it but it looks right):

```
def stub_features(hash)
  Feature.features_hash = nil
  @original_features = Feature.method(:raw_features)
  Feature.define_singleton_method(:raw_features) { Rails.application.config.options.features.merge(hash) }
  Rails.application.reload_routes!
end

def unstub_features
  Feature.singleton_class.send(:remove_method, :raw_features)
  Feature.define_singleton_method(:raw_features, @original_features)
  Feature.features_hash = nil
  Rails.application.reload_routes!
end

def stub_settings(hash)
  @original_settings = Setting.method(:settings)
  Setting.define_singleton_method(:settings) { Rails.application.config.options.settings.merge(hash) }
  Rails.application.reload_routes!
end

def unstub_settings
  Setting.singleton_class.send(:remove_method, :settings)
  Setting.define_singleton_method(:settings, @original_settings)
  Rails.application.reload_routes!
end
```

2) Then the before can be modified to:

```
setup do
  stub_features(feature1: true, feature2: false)
  stub_settings(setting1: 'value1', setting2: 'value2')
end

teardown do
  unstub_features
  unstub_settings
end
```

This way we are confident that every test is running with the feature fully enabled, so we're not inadvertently short-circuiting any code.